File: 03-Stack/tldusqkr/B-10828.py
import sys

stack = list()
top = len(stack)-1

def push(num):
    stack.append(num)
def pop():
    try:
        print(stack.pop())
    except:
        print(-1)
def size():
    return(len(stack))
def empty():
    result = 1 if size() == 0 else 0
    return result
def top():
    try:
        print(stack[-1])
    except:
        print(-1)

N = int(sys.stdin.readline())

for _ in range(N):
    cmd = sys.stdin.readline().split()
    if cmd[0] == 'push':
        push(cmd[1])
    elif cmd[0] == 'pop':
        pop()
    elif cmd[0] == 'empty':
        print(empty())
    elif cmd[0] == 'size':
        print(size())
    elif cmd[0] == 'top':
        top()

File: 03-Stack/holyo/B-10828.cpp
#include <iostream>
#include <stack>
#include <string>

using namespace std;

int main() {
    int n;
    cin >> n;
    stack<int> a;
    while (n--) {
        string cmd;
        cin >> cmd;
        if (cmd == "push"){
            int num;
            cin >> num;
            a.push(num);
        }
        else if (cmd == "top") {
            cout << (a.empty() ? -1 : a.top()) << endl;
        }
        else if (cmd == "size") {
            cout << a.size() << endl;
        }
        else if (cmd == "empty") {
            cout << a.empty() << endl;
        }
        else if (cmd == "pop") {
            cout << (a.empty() ? -1 : a.top()) << endl;
            if (!a.empty()) {
                a.pop();
            }
        }
    }
    return 0;
}
